### ISSUE TITLE
Updated Constants for GDBM_File

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#3663]
Migrated from [rt.perl.org#6339](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=6339) (status was 'resolved')

Searchable as RT6339$

**Comment by p5pRT** [comment]
### From plindner@localhost.localdomain


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
GDBM\_File included in Perl 5\.6 is missing a constant added in the gdbm\-1\.8\.\* 
versions&#8203;: GDBM\_NOLOCK

I've patched up the xs and pm to do this&#8203;:


\*\*\* GDBM\_File\.pm\.orig    Mon Mar 12 03&#8203;:36&#8203;:09 2001
\-\-\- GDBM\_File\.pm    Mon Mar 12 03&#8203;:36&#8203;:34 2001
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 57\,65 \*\*\*\*
&nbsp;      GDBM\_REPLACE
&nbsp;      GDBM\_WRCREAT
&nbsp;      GDBM\_WRITER
&nbsp;  \);
&nbsp;  
\! $VERSION = "1\.03";
&nbsp;  
&nbsp;  sub AUTOLOAD \{
&nbsp;      my\($constname\);
\-\-\- 57\,66 \-\-\-\-
&nbsp;      GDBM\_REPLACE
&nbsp;      GDBM\_WRCREAT
&nbsp;      GDBM\_WRITER
\+     GDBM\_NOLOCK
&nbsp;  \);
&nbsp;  
\! $VERSION = "1\.04";
&nbsp;  
&nbsp;  sub AUTOLOAD \{
&nbsp;      my\($constname\);



\*\*\* GDBM\_File\.xs\.orig    Mon Mar 12 03&#8203;:36&#8203;:01 2001
\-\-\- GDBM\_File\.xs    Mon Mar 12 03&#8203;:33&#8203;:49 2001
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 146\,151 \*\*\*\*
\-\-\- 146\,161 \-\-\-\-
&nbsp;  \#else
&nbsp;          goto not\_there;
&nbsp;  \#endif
\+ 
\+     if \(strEQ\(name\, "GDBM\_NOLOCK"\)\)
\+ \#ifdef GDBM\_NOLOCK
\+         return GDBM\_NOLOCK;
\+ \#else
\+         goto not\_there;
\+ \#endif
\+ 
\+ L1
\+ 
&nbsp;      break;
&nbsp;      case 'H'&#8203;:
&nbsp;      break;

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=medium

Site configuration information for perl v5.6.0:

Configured by prospector at Mon Aug  7 10:58:30 EDT 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.2.5-22smp, archname=i386-linux
    uname='linux porky.devel.redhat.com 2.2.5-22smp #1 smp wed jun 2 09:11:51 edt 1999 i686 unknown '
    config_args='-des -Doptimize=-O2 -march=i386 -mcpu=i686 -Dcc=gcc -Dcccdlflags=-fPIC -Dinstallprefix=/usr -Dprefix=/usr -Darchname=i386-linux -Dd_dosuid -Dd_semctl_semun -Di_db -Di_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Uuselargefiles'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=undef 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='gcc', optimize='-O2 -march=i386 -mcpu=i686', gccversion=2.96 20000731 (experimental)
    cppflags='-fno-strict-aliasing'
    ccflags ='-fno-strict-aliasing'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -ldl -lm -lc -lcrypt
    libc=/lib/libc-2.1.92.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /usr/lib/perl5/5.6.0/i386-linux
    /usr/lib/perl5/5.6.0
    /usr/lib/perl5/site_perl/5.6.0/i386-linux
    /usr/lib/perl5/site_perl/5.6.0
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/plindner
    LANG=en_US
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/kerberos/bin:/bin:/usr/bin:/usr/bin/X11:/usr/local/bin:/usr/bin:/usr/X11R6/bin:/opt/oracle/product/8.1.6/bin:/home/plindner/bin:/opt/oracle/product/8.1.6/bin:/home/plindner/bin:/opt/oracle/product/8.1.6/bin:/home/plindner/bin:/opt/oracle/product/8.1.6/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From plindner@localhost.localdomain


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
GDBM\_File included in Perl 5\.6 is missing a constant added in the gdbm\-1\.8\.\* 
versions&#8203;: GDBM\_NOLOCK

I've patched up the xs and pm to do this&#8203;:


\*\*\* GDBM\_File\.pm\.orig    Mon Mar 12 03&#8203;:36&#8203;:09 2001
\-\-\- GDBM\_File\.pm    Mon Mar 12 03&#8203;:36&#8203;:34 2001
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 57\,65 \*\*\*\*
&nbsp;      GDBM\_REPLACE
&nbsp;      GDBM\_WRCREAT
&nbsp;      GDBM\_WRITER
&nbsp;  \);
&nbsp;  
\! $VERSION = "1\.03";
&nbsp;  
&nbsp;  sub AUTOLOAD \{
&nbsp;      my\($constname\);
\-\-\- 57\,66 \-\-\-\-
&nbsp;      GDBM\_REPLACE
&nbsp;      GDBM\_WRCREAT
&nbsp;      GDBM\_WRITER
\+     GDBM\_NOLOCK
&nbsp;  \);
&nbsp;  
\! $VERSION = "1\.04";
&nbsp;  
&nbsp;  sub AUTOLOAD \{
&nbsp;      my\($constname\);



\*\*\* GDBM\_File\.xs\.orig    Mon Mar 12 03&#8203;:36&#8203;:01 2001
\-\-\- GDBM\_File\.xs    Mon Mar 12 03&#8203;:33&#8203;:49 2001
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 146\,151 \*\*\*\*
\-\-\- 146\,161 \-\-\-\-
&nbsp;  \#else
&nbsp;          goto not\_there;
&nbsp;  \#endif
\+ 
\+     if \(strEQ\(name\, "GDBM\_NOLOCK"\)\)
\+ \#ifdef GDBM\_NOLOCK
\+         return GDBM\_NOLOCK;
\+ \#else
\+         goto not\_there;
\+ \#endif
\+ 
\+ L1
\+ 
&nbsp;      break;
&nbsp;      case 'H'&#8203;:
&nbsp;      break;

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=medium

Site configuration information for perl v5.6.0:

Configured by prospector at Mon Aug  7 10:58:30 EDT 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.2.5-22smp, archname=i386-linux
    uname='linux porky.devel.redhat.com 2.2.5-22smp #1 smp wed jun 2 09:11:51 edt 1999 i686 unknown '
    config_args='-des -Doptimize=-O2 -march=i386 -mcpu=i686 -Dcc=gcc -Dcccdlflags=-fPIC -Dinstallprefix=/usr -Dprefix=/usr -Darchname=i386-linux -Dd_dosuid -Dd_semctl_semun -Di_db -Di_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Uuselargefiles'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=undef 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='gcc', optimize='-O2 -march=i386 -mcpu=i686', gccversion=2.96 20000731 (experimental)
    cppflags='-fno-strict-aliasing'
    ccflags ='-fno-strict-aliasing'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -ldl -lm -lc -lcrypt
    libc=/lib/libc-2.1.92.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /usr/lib/perl5/5.6.0/i386-linux
    /usr/lib/perl5/5.6.0
    /usr/lib/perl5/site_perl/5.6.0/i386-linux
    /usr/lib/perl5/site_perl/5.6.0
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/plindner
    LANG=en_US
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/kerberos/bin:/bin:/usr/bin:/usr/bin/X11:/usr/local/bin:/usr/bin:/usr/X11R6/bin:/opt/oracle/product/8.1.6/bin:/home/plindner/bin:/opt/oracle/product/8.1.6/bin:/home/plindner/bin:/opt/oracle/product/8.1.6/bin:/home/plindner/bin:/opt/oracle/product/8.1.6/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>
